drivers/platform/surface/surface_fan.c

```diff
@@ -0,0 +1,146 @@
+// SPDX-License-Identifier: GPL-2.0+
+/*
+ * Surface Fan driver for Surface System Aggregator Module. It provides access
+ * to the fan's rpm through the hwmon system.
+ *
+ * Copyright (C) 2023 Ivor Wanders <ivor@iwanders.net>
+ */
+
+#include <linux/hwmon.h>
+#include <linux/kernel.h>
+#include <linux/module.h>
+#include <linux/platform_device.h>
+#include <linux/surface_aggregator/device.h>
+#include <linux/types.h>
+
+#define SURFACE_FAN_MIN_SPEED 2000
+#define SURFACE_FAN_MAX_SPEED 7500
+
+struct fan_data {
+	struct device *dev;
+	struct ssam_device *sdev;
+	struct device *hdev;
+};
+
+// SSAM
+SSAM_DEFINE_SYNC_REQUEST_R(__ssam_fan_get, __le16,
+			   {
+				   .target_category = SSAM_SSH_TC_FAN,
+				   .target_id = SSAM_SSH_TID_SAM,
+				   .command_id = 0x01,
+				   .instance_id = 0x01,
+			   });
```

Comment:
I'd prefer if this would be formatted like a struct, not a function call (i.e. remove the indents and move the bracket, see e.g. [here](https://github.com/linux-surface/kernel/blob/8cd23b1bb3a8b7a3ef7cec2c37e7e46e6397a858/drivers/platform/surface/surface_platform_profile.c#L35-L38)).

Also, I think this can be made a [`SSAM_DEFINE_SYNC_REQUEST_CL_R`](https://elixir.bootlin.com/linux/latest/C/ident/SSAM_DEFINE_SYNC_REQUEST_CL_R), which will define a function to take in the `ssam_device` instead of the controller. So you only need to specify the `target_category` and the `command_id`, with the rest coming from the device you pass in. That way, if we ever have multiple fans (which I assume will be separated by instance ID), we only need to register one device for each fan in the registry and the rest (including this request) will work automatically.

Reply:
Interesting, this is what clang format turned it into. Guess I'm lucky that at work we just accept whatever clang format does :D 

With 6c14d0e336062294e6d694321431e27ad59d2f16 I changed the formatting and use the `_CL_R`, I've also updated the name of the function to `__ssam_fan_rpm_get` to ensure the `rpm` is in the name.